bridge/engine/EngineMixer.cpp

```diff
@@ -1202,27 +1295,59 @@ void EngineMixer::checkInboundPacketCounters(const uint64_t timestamp)
 
 void EngineMixer::checkRecordingPacketCounters(const uint64_t timestamp)
 {
+    // We will give 15 seconds before delete outbound context since it was marked for deletion
+    // because we can still receive some NACK or need to resend some recording events if smb hasn't
+    // receive the ACKs. The recording events are especially important to recording to be converted properly
+    static constexpr uint64_t decommissionedTimeout = 15 * utils::Time::sec;
     for (auto& recordingStreamEntry : _engineRecordingStreams)
     {
         const auto endpointIdHash = recordingStreamEntry.first;
         for (auto& outboundContextEntry : recordingStreamEntry.second->ssrcOutboundContexts)
         {
             auto& outboundContext = outboundContextEntry.second;
-            if (outboundContext.markedForDeletion &&
-                utils::Time::diffGT(outboundContext.lastSendTime, timestamp, utils::Time::sec * 30))
+
+            const bool isDecommissioned =
+                outboundContext.recordingOutboundDecommissioned && !outboundContext.markedForDeletion;
+
+            if (isDecommissioned && utils::Time::diffGT(outboundContext.lastSendTime, timestamp, decommissionedTimeout))
             {
-                logger::info("Removing marked and idle recording outbound context ssrc %u, rec endpointIdHash %lu",
+                logger::info("Marking for deletion decommissioned recording outbound context ssrc %u, rec "
+                             "endpointIdHash %zu",
                     _loggableId.c_str(),
                     outboundContextEntry.first,
                     endpointIdHash);
 
-                EngineMessage::Message message(EngineMessage::Type::FreeRecordingRtpPacketCache);
-                message.command.freeRecordingRtpPacketCache.mixer = this;
-                message.command.freeRecordingRtpPacketCache.ssrc = outboundContext.ssrc;
-                message.command.freeRecordingRtpPacketCache.endpointIdHash = recordingStreamEntry.first;
-                _messageListener.onMessage(std::move(message));
+                const auto ssrc = outboundContextEntry.first;
+                outboundContext.markedForDeletion = true;
+                auto& outboundContextsFinalizerCounter = recordingStreamEntry.second->outboundContextsFinalizerCounter;
+                auto counterIt = outboundContextsFinalizerCounter.find(ssrc);
```

Comment:
This is a quite complex, EngineRecordingStream can have multiple transports (redundancy feature that we don't use yet and don't know if we will use it). But that create complex on syncronization has we need to ensure that all transports have executed the finalised before remove the outbound context.

outboundContextsFinalizerCounter is a stores a atomic counter of finalizers to know we have run all finalize jobs before remove the context